hooks/pre-command

```diff
@@ -136,17 +153,50 @@ if [[ ! -d "${unpack_dest}" || "${orig_mtime}" != "${post_dl_mtime}" ]]; then
     echo "New or updated installation detected, unpacking!"
     rm -rf "${unpack_dest}"
     mkdir -p "${unpack_dest}"
-    tar -xof "${tarball_path}" --strip-components=1 -C "${unpack_dest}"
+    if [[ "${archive_path}" == *.tar.gz ]]; then
+      tar -xof "${archive_path}" --strip-components=1 -C "${unpack_dest}"
+    elif [[ "${archive_path}" == *.zip ]]; then
+      unzip -q "${archive_path}" -d "${unpack_dest}"
+      mv "${unpack_dest}"/julia-*/* "${unpack_dest}"
+      rmdir "${unpack_dest}"/julia-*
+    else
+      buildkite-agent annotate --style error "Unhandled archive type ${archive_path}"
+      exit 1
+    fi
 fi
-# Export `PATH` such that this `julia` is the first one other plugins find
-export PATH=${unpack_dest}/bin:$PATH
 
 # Show that julia is installed and print out the version info
 echo "--- :julia: Installation Successful"
-echo "Julia ${JULIA_VERSION} successfully installed to $(which julia)"
-julia -e 'using InteractiveUtils; versioninfo()'
+echo "Julia ${JULIA_VERSION} successfully installed to ${unpack_dest}"
+"${unpack_dest}/bin/julia" -e 'using InteractiveUtils; versioninfo()'
 
+# Export `PATH` such that this `julia` is the first one other plugins find
+if [[ "$OSTYPE" == "linux"* ]]; then
+  export PATH=${unpack_dest}/bin:"$PATH"
+elif [[ "$OSTYPE" == "msys" ]]; then
+  # modifying the environment on Windows doesn't work, not by setting the
+  # PATH env var (picked up by a Buildkite pre-exit hook, but subsequently
+  # setting an incomplete environment later on, failing to find bash.exe again),
+  # or even by directly modifying HKCU\Environment:Path (which somehow does not
+  # get picked up by subsequently-executed Buildkite plugins.
+  #
+  # HACK: generate a wrapper script and put it in a location that really ought
+  #       to be part of the environment (i.e., the buildkite-agent itself).
+  ## first stage is wrapper called `julia`, sourcing an agent-specific script
+  wrapper=$(dirname $(which buildkite-agent))/julia
```

Comment:
This may break hosting multiple runners from the same machine, because we're modifying global state here.

Reply:
See the bits below, the global wrapper is always the same, the second-stage one is local to the agent.

Reply:
Oh, I see.  My bad, I didn't read carefully enough